Issue440 properties page work

DaveTrost opened the pull request:
### What issue is this solving?
Addresses issue #440

### Any helpful knowledge/context for the reviewer?
Is a `npm install` necessary? No
Any special requirements to test? No

### Please make sure you've attempted to meet the following coding standards
- [x] Code builds successfully
- [x] Code has been tested and does not produce errors
- [x] Code is readable and formatted
- [x] There isn't any unnecessary commented-out code
- [x] There aren't any unnecessary commits or files changed (shouldn't touch the /stashed folder unless the issue requests it)

If you're having trouble meeting this criteria, feel free to reach out on Slack for help!

DaveTrost commented:
I opened a backend PR related to the functionality in this branch: codeforpdx/dwellinglybackend/pull/166
Ideally, the backend PR can be closed before this one so I can add to this branch with the code that makes a request to the new API route

Chris-Boe commented on src/components/ShowHideSwitch/index.js:
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import './showHideSwitch.scss';
+
+export const ShowHideSwitch = ({ id, labelText, isShowState, handleToggleChange }) => (
```
This switch looks awesome! Great job

Chris-Boe commented on src/views/properties/properties.js:
```diff
@@ -4,12 +4,22 @@ import UserContext from '../../UserContext';
   dataField: 'name',
+  formatter: (_, row) => {
+    return (
+      <Link key={row.id} to={`/manage/property/${row.id}`}>
```
The selectRow functionality seems to be overriding this Link functionality as I can't click into the property details page using the property name

DaveTrost replied:
This part is going to be weird until we have a view/edit properties page

DaveTrost replied:
We have a view/edit properties page - hooray! Thank you @Protopaco!
Clicking on the property name now works as expected. Please take a look @Chris-Boe

Chris-Boe reviewed (changes requested):
Just one issue when trying to click the property name to see the property details page.
After a Slack convo it sounds like we will add the actual archive API calling to this PR after the backend PR is merged

DaveTrost commented:
Still waiting for codeforpdx/dwellinglybackend/pull/166 to be merged so this PR can go live. Ping me on Slack if you notice anything (or if this branch needs a rebase)

DaveTrost commented:
> Still waiting for [codeforpdx/dwellinglybackend/pull/166](https://github.com/codeforpdx/dwellinglybackend/pull/166) to be merged so this PR can go live. Ping me on Slack if you notice anything (or if this branch needs a rebase)

The backend PR is merged. This PR is ready to go! 
@Chris-Boe @htharker42

Chris-Boe reviewed (changes requested):
Nice work Dave! I'm just noticing one issue with the search/archive filtering. If I have the Show Archived toggle set to false, type in a search that would match an archived property, then toggle Show Archived to *on*, I still don't see the property that would match my search criteria:
![search](https://user-images.githubusercontent.com/8600792/102725804-56aff380-42ce-11eb-98da-b22d4ffc9448.gif)

DaveTrost committed:
Add action button for "Archive Properties"

By default, the button is disabled and it only becomes active when one or more properties have their checkboxes selected.
After the archive request is complete, reset the table element and update the list on the page
New state tracks the selected properties.
archivedCallCount is a workaround to force a re-render on the bootstrap table
css cleanup
selectRow object needs to have access to state, so it moves within the scope of the component.

DaveTrost committed:
apply sort direction fix to the tickets page, based on learnings from view properties page

DaveTrost committed:
set default sort order as "ascending" (A to Z)

DaveTrost committed:
functional component refactor!

Maintain consistent table sizing even when no rows are displayed - use min-width

DaveTrost committed:
better interaction between search bar and "show archived"

Refactor for better variable naming, grouping of related functions, add helper function `getDisplayProperties`

DaveTrost committed:
New component showHideSwitch built and added to properties page to controls display of archived properties

New state variables and event handlers in `Properties` component to enable filtering of archived properties
Styling to incorporate the show/hide toggle into the Properties List page
The search bar was not wide enough for the placeholder message, and some CSS tweaks were needed. Other components can limit the width of the search box at the container level if needed

DaveTrost committed:
connect Archive Properties button to the back end.

refactoring:
- more descriptive variable naming in state
- reusable function setTableDataFromProperties

DaveTrost committed:
search text remains active before and after "Archive Properties"

DaveTrost commented:
> Nice work Dave! I'm just noticing one issue with the search/archive filtering. If I have the Show Archived toggle set to false, type in a search that would match an archived property, then toggle Show Archived to _on_, I still don't see the property that would match my search criteria:
> ![search](https://user-images.githubusercontent.com/8600792/102725804-56aff380-42ce-11eb-98da-b22d4ffc9448.gif)

Thank you @Chris-Boe for the helpful screen capture! It made it super simple to see what you were highlighting. 
This issue is resolved. Please let me know if you discover anything else ... which you won't because it is now BULLET-PROOF! Muahahaha

htharker42 reviewed (approved):
is working now, good job Dave
